### DISCUSSION TITLE
[Bug Report] Crash when updating feeds

### DISCUSSION THREAD

**Discussion by sophie-la-li** [spacecowboy/Feeder#156] [General]
Hi. So i update my feeds manually. And if i do so and there are a lot of new messages to load, 
- the app lags
- then it crashes
- and then i have a lot (!) email drafts

It seems that this app has a serious issue with memory. I never developed for android and in Java/Kotlin, so i can't easily look up the code, but there should be some kind of garbage cleanup in the process?!

Can you please give us an option to disable the email draft spam?

```
Application: com.nononsenseapps.feeder (flavor release)
Version: 2.6.11 (code 296)
Android: 14 (SDK 34)
Device: Google Pixel 5

Unhandled exception:

java.lang.OutOfMemoryError: Failed to allocate a 16 byte allocation with 824928 free bytes and 805KB until OOM, target footprint 268435456, growth limit 268435456; giving up on allocation because <1% of heap free after GC.
at androidx.compose.runtime.CompositionImpl.addPendingInvalidationsLocked(SourceFile:5)
at androidx.compose.runtime.CompositionImpl.drainPendingModificationsLocked(Unknown Source:22)
at androidx.compose.runtime.Recomposer.access$recordComposerModificationsLocked(SourceFile:112)
at androidx.compose.material3.CheckboxKt$CheckboxImpl$1$1.invoke(SourceFile:542)
at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame(Unknown Source:6)
at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame(SourceFile:48)
at android.view.Choreographer$[CallbackRecord.run](https://callbackrecord.run/)(choreographer.java:1337)
at android.view.Choreographer$[CallbackRecord.run](https://callbackrecord.run/)(choreographer.java:1348)
at android.view.Choreographer.doCallbacks(choreographer.java:952)
at android.view.Choreographer.doFrame(choreographer.java:878)
at android.view.Choreographer$[FrameDisplayEventReceiver.run](https://framedisplayeventreceiver.run/)(choreographer.java:1322)
at android.os.Handler.handleCallback(handler.java:958)
at android.os.Handler.dispatchMessage(handler.java:99)
at android.os.Looper.loopOnce(looper.java:205)
at android.os.Looper.loop(looper.java:294)
at [android.app](https://android.app/).ActivityThread.main(activitythread.java:8177)
at java.lang.reflect.Method.invoke(Native Method)
at [com.android](https://com.android/).internal.os.RuntimeInit$[MethodAndArgsCaller.run](https://methodandargscaller.run/)(runtimeinit.java:552)
at [com.android](https://com.android/).internal.os.ZygoteInit.main(zygoteinit.java:971)
Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [androidx.compose.runtime.PausableMonotonicFrameClock@bbae5e0, androidx.compose.ui.platform.MotionDurationScaleImpl@bde5099, StandaloneCoroutine{Cancelling}@98d1f5e, AndroidUiDispatcher@1592f3f]
```

Messages to load in this case: ~150
Emails created: 11
Feeds: 59

**Comment by spacecowboy** [comment]
The email draft is not spam. It is how you should report crashes. Please send the email.

Memory issues are difficult to diagnose because they depend on your specific phone and what settings you are using.

**Reply by sophie-la-li** [reply]
If it would be one email i would not argue, but it's a lot of them at once. All with the same content. I sent one of them.

The device is a Google Pixel 5 using Android 14. The options (i think are relevant) are:
- only manual update
- 50 per feed
- no device sync
- show preview images
- don't load full articles

Right now without feeder doing anything i have 4gb of memory left. A lot needs to happen to exceed that.

**Comment by sophie-la-li** [comment]
[feeder-export-2023-12-18T11_58_18.334.opml.zip](https://github.com/spacecowboy/Feeder/files/13702841/feeder-export-2023-12-18T11_58_18.334.opml.zip)

**Reply by sophie-la-li** [reply]
I tested removing all caches and data and reimporting feeds and settings. Afterwards everything seemed to work flawlessly. But shortly after i moved some feeds into another group the issues came up again.

Don't know if there might be connection or if it's just a coincidence.

**Reply by spacecowboy** [reply]
Thank you for investigating a bit. Will be helpful.

So to summarize

* Open the app
* Import OMPL and let it sync
* Change tags of a couple of feeds
* Sync again
* Out of memory happens

is a decent summary of the flow?

None of you feeds have a "Fetch full article by default" set?

I don't suppose you've ever had automatic sync active and seen it crash in the background? Trying to understand if it is the sync code consuming the memory or if it's the UI.

**Reply by sophie-la-li** [reply]
I try to get some concrete steps to reproduce the issue, but was not able to, yet. It's erratic. It seems that this involves a step like "leave the app and wait for some hours until there are new news to fetch", but I'm unsure here. I'll try and come back if i have found a way to reproduce it.

What i can see is that the app is using up to 500mb each time i trigger an update. But i don't know what's low or high in that case. As said I'm not a android dev.

PS: No i never used the sync.

**Reply by spacecowboy** [reply]
500mb is on the insane end of the scale.

good number would be less than 10 MB in my opinion

**Reply by sophie-la-li** [reply]
![Screenshot_20231218-184852](https://github.com/spacecowboy/Feeder/assets/6147076/251d91a2-e642-47eb-85af-23b4b0af771b)

Hope you can read that even though it's German.

**Reply by spacecowboy** [reply]
Strange. Mine is only at 18MB average. And I have a ton of feeds.

![Screenshot_20231218_185739_Settings](https://github.com/spacecowboy/Feeder/assets/223655/52a8c3a6-4e39-4686-bff4-0624b85fb6df)

**Reply by spacecowboy** [reply]
500MB is a lot, but should not be enough to crash your phone. Unless something else is taking up all the memory (and that's the real problem)

**Reply by sophie-la-li** [reply]
![Screenshot_20231218-194736](https://github.com/spacecowboy/Feeder/assets/6147076/a8f45fd1-66f4-444c-931b-a43008302e0f)

Well right now there is enough memory left. OC i can't tell how the situation is when the crash occurs. But that crash only affects the Feeder app. Everything else is still running fine.

**Reply by sophie-la-li** [reply]
https://developer.android.com/topic/performance/memory#CheckHowMuchMemory

According to this, " Android sets a hard limit on the heap size allotted for each app ". I have a feeling that this might be 512mb?

**Comment by sophie-la-li** [comment]
Not happened again since month now. I would consider this fixed somehow.

**Reply by spacecowboy** [reply]
Cool.

I changed how feeds are parsed. This was always high on my suspects for memory leaks.

Good to know it's working better now :+1:
